src/scopyApplication.cpp

```diff
@@ -0,0 +1,99 @@
+#include <QApplication>
+#include "scopyApplication.hpp"
+#include <QDir>
+#include <QFileInfo>
+#include <QDateTime>
+#include <QDebug>
+
+using namespace google_breakpad;
+
+ScopyApplication::ScopyApplication(int& argc, char** argv) : QApplication(argc, argv) {}
+QString ScopyApplication::initBreakPadHandler(QString crashDumpPath) {
+	QString prevCrashDump="";
+	QString appDir = crashDumpPath;
+	QDir qd(appDir);
+	if(qd.exists()) {
+		qd.mkdir("crashdmp");
+	}
+	qd.cd("crashdmp");
+	QStringList dumps= qd.entryList();
+	for(QString dump : dumps)
+	{
+		if(dump=="." || dump == "..") continue;
+		if(dump.startsWith("ScopyCrashDump"))
+			continue;
+		QString dumpFullPath = qd.path()+qd.separator()+dump;
+		QFileInfo fi(dumpFullPath);
+		QString dumpDateAndTime = fi.lastModified().toString(Qt::ISODate).replace("T","--").replace(":","-");
+		prevCrashDump = qd.path()+qd.separator()+"ScopyCrashDump--"+dumpDateAndTime+".dmp";
+		QFile::rename(dumpFullPath, prevCrashDump);
+
+
+	}
+
+	QByteArray ba = qd.path().toLocal8Bit();
+	const char *appDirStr = ba.data();
+
+#ifdef Q_OS_LINUX
+	descriptor = new google_breakpad::MinidumpDescriptor(qd.path().toStdString().c_str());

```

Comment:
You could use unique pointer here. This way you can avoid the #ifdef q_os_linux in destructor and even skip defining the desctructor.

Reply:
I'd rather not use complex objects (such as unique_ptr) when dealing with these kinds of exceptions as they operate very low level.

Reply:
ok